src/main/java/de/unistuttgart/memorybackend/service/ConfigService.java

```diff
@@ -0,0 +1,206 @@
+import de.unistuttgart.memorybackend.data.mapper.CardPairMapper;
+import de.unistuttgart.memorybackend.data.mapper.ConfigurationMapper;
+import de.unistuttgart.memorybackend.repositories.CardPairRepository;
+import de.unistuttgart.memorybackend.repositories.ConfigurationRepository;
+import jakarta.annotation.PostConstruct;
+import java.util.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+@ComponentScan("de.unistuttgart.memorybackend.data.mapper")
+@Service
+@Transactional
+public class ConfigService {
+
+    @Autowired
+    CardPairMapper cardPairMapper;
+
+    @Autowired
+    ConfigurationMapper configurationMapper;
+
+    @Autowired
+    ConfigurationRepository configurationRepository;
+
+    @Autowired
+    CardPairRepository cardPairRepository;
+
+    /**
+     * Search a configuration by given id
+     *
+     * @param id the id of the configuration searching for
+     * @return the found configuration
+     * @throws ResponseStatusException  when configuration by configurationName could not be found
+     * @throws IllegalArgumentException if at least one of the arguments is null
+     */
+    public Configuration getConfiguration(final UUID id) {
+        if (id == null) {
+            throw new IllegalArgumentException("id is null");
+        }
+        return configurationRepository
+            .findById(id)
+            .orElseThrow(() ->
+                new ResponseStatusException(
+                    HttpStatus.NOT_FOUND,
+                    String.format("There is no configuration with id %s.", id)
+                )
+            );
+    }
+
+    /**
+     * Save a configuration
+     *
+     * @param configurationDTO configuration that should be saved
+     * @return the saved configuration as DTO
+     * @throws IllegalArgumentException if at least one of the arguments is null
+     */
+    public ConfigurationDTO saveConfiguration(final ConfigurationDTO configurationDTO) {
+        if (configurationDTO == null) {
+            throw new IllegalArgumentException("configurationDTO is null");
+        }
+        final Configuration savedConfiguration = configurationRepository.save(
+            configurationMapper.configurationDTOToConfiguration(configurationDTO)
+        );
+        return configurationMapper.configurationToConfigurationDTO(savedConfiguration);
+    }
+
+    /**
+     * Update a configuration
+     *
+     * @param id               the id of the configuration that should be updated
+     * @param configurationDTO configuration that should be updated
+     * @return the updated configuration as DTO
+     * @throws ResponseStatusException  when configuration with the id does not exist
```

Comment:
```suggestion
     * @throws ResponseStatusException (404) when configuration with the id does not exist
```
Just mark on the ResponseStatusException what status code will be thrown. I will not comment on the rest.